core/src/test/scala/org/apache/spark/util/ResetSystemProperties.scala

```diff
@@ -42,7 +43,7 @@ private[spark] trait ResetSystemProperties extends BeforeAndAfterEach { this: Su
   var oldProperties: Properties = null
 
   override def beforeEach(): Unit = {
-    oldProperties = new Properties(System.getProperties)
+    oldProperties = SerializationUtils.clone(System.getProperties)
```

Comment:
thanks for tracking this down.   can you please put a comment in here explaining why you need a clone?  it is really subtle, I can easily see this getting reverted down the road if somebody doesn't know why its there.